stdlib/REPL/test/repl.jl

```diff
@@ -914,15 +914,25 @@ let ends_with_semicolon = REPL.ends_with_semicolon
     @test !ends_with_semicolon("")
     @test ends_with_semicolon(";")
     @test !ends_with_semicolon("a")
+    @test !ends_with_semicolon("a # asdf ;")
+    @test ends_with_semicolon("""a * "#" ;""")
+    @test ends_with_semicolon("a; #=#=# =# =#\n")
     @test ends_with_semicolon("1;")
     @test ends_with_semicolon("1;\n")
     @test ends_with_semicolon("1;\r")
     @test ends_with_semicolon("1;\r\n   \t\f")
     @test ends_with_semicolon("1;#text\n")
-    @test ends_with_semicolon("a; #=#=# =# =#\n")
+    @test ends_with_semicolon("""1;\n#text\n""")
+    @test ends_with_semicolon("1; #=# 2")
```

Comment:
This test should be removed or modified — self-closing comments `#=#` were removed from Julia way back in 2014 (#6444), so this is invalid syntax.

Similarly, the other tests that use the `#=#` pattern should be modified.

Reply:
In particular, this gives a parse error `syntax: incomplete: unterminated multi-line comment #= ... =#`. 

So we should give *some* answer for this string (we shouldn't have an infinite loop or an exception), but it doesn't really matter *what* answer we give.